recipe/meta.yaml

```diff
@@ -26,12 +26,12 @@ requirements:
     - numpy >=1.21.0,<2.0.0
```

Comment:
Hi @BastianZim we didn't explicitly cap numpy below 2 if that's what you're referring to, but we did want to allow newer Python versions (this is why change to `python >=3.7.1,<4.0`) and unfortunately older numpy versions explicitly disallow Python 3.11, thus a following condition in `pyproject.toml`:
```
numpy = [
    { version = ">=1.21.0 <1.23.4", markers = "python_full_version >= '3.7.1' and python_full_version < '3.11.0'" },
    { version = "^1.23.4", markers = "python_full_version >= '3.11.0'" },
]
```
I think it's reasonable to limit numpy to a major version though so IMO it's fine to have it as-is but happy to hear your recommendations

Reply:
@BastianZim Ignore the above comment -- yes I think it's reasonable to rely on solver here. I've removed numpy from the list. Thanks!

Reply:
Ok, but now you removed NumPy completely, it should still be there but with all possible versions. 
Basically, what I meant is the following:
You specified different NumPy versions for different python versions which makes sense for pip because of the reasons you outlined, but in conda, the solver will match the right NumPy version to the right python version.
Therefore, in conda, the absolute minimum version (>=1.21.0) and the absolute maximum version (<2) should be used and the solver will make sure that 1.23.4 is not used when running under pyhton3.11.
I just wanted to make sure that the reason you listed different NumPy versions is only because of python and not something else that I overlooked. :)
So, in short, my approach is fine, if it's only because of python, otherwise we need to limit it more strictly.

Reply:
If you're fine with it, let me push my version and then you can check if it can be merged from your side?

Reply:
Ah I see. Yes it's only because of python - poetry install wasn't able to figure out the correct version otherwise and would complain about inability to support 3.11. Ok, let me bring back numpy -- but do I understand correctly that `pip check` still has to go? Because it keeps failing irrespective of numpy

Reply:
Ok, perfect. Yes, pip check needs to go because it isn't correctly matching it. Sometimes conda-forge allows for more combinations than pip check, so that's not always working. If you still want to keep it, we can also patch it locally. Let me know if you want to have it or not?

Reply:
Yes it would be nice to have as an extra safety check :)

Reply:
Ok I'm not on my computer but will patch it out later.

Reply:
Sounds good, thanks for help!

Reply:
Ok let's see if that works. 

Also ordered the deps alphabetically so that Grayskull can try updating the next time.

Reply:
OK should be fine. Please double check the deps and we can merge.

Reply:
Awesome! Thanks so much!